src/System.Net.Http/src/System/Net/Http/DiagnosticsHandler.cs

```diff
@@ -0,0 +1,92 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace System.Net.Http
+{
+    /// <summary>
+    /// DiagnosticHandler notifies DiagnosticSource subscribers about outgoing Http requests
+    /// </summary>
+    internal sealed class DiagnosticsHandler : DelegatingHandler
+    {
+        /// <summary>
+        /// DiagnosticHandler constructor
+        /// </summary>
+        /// <param name="innerHandler">Inner handler: Windows or Unix implementation of HttpMessageHandler. 
+        /// Note that DiagnosticHandler is the latest in the pipeline </param>
+        public DiagnosticsHandler(HttpMessageHandler innerHandler) : base(innerHandler)
+        {
+        }
+
+        internal static bool IsEnabled()
+        {
+            return s_diagnosticListener.IsEnabled(DiagnosticsHandlerLoggingStrings.ResponseWriteName);
```

Comment:
Our assumption is that Response is the most interesting event, so there may be scenarios when user is interested in Response only or in both: Request and Response.
All subscribers I'm aware of, including Intellitrace, subscribe for both events.
Do you have any particular scenario for pure Request event?